Kernel/Devices/Device.h

```diff
@@ -75,7 +75,7 @@ class Device : public File {
         {
             ScopedSpinLock lock(m_requests_lock);
             was_empty = m_requests.is_empty();
-            m_requests.append(request);
+            m_requests.append(RefPtr<AsyncDeviceRequest>(request));
```

Comment:
This change feels like an ergonomics regression. Can we avoid forcing clients to do this?

Reply:
Yes. I was being overly prescriptive, I guess. The `static_assert` in `DoublyLinkedList`'s constructor and `append()` were requiring the type being passed in to be the same as the type in the container. This is not really required though, as long as it is convertible. I will change that `static_assert` to check for conversion rather than same-ness.

The `static_assert` *could* be removed completely, but in doing so I think the compiler errors might get obtuse. My reason for adding the `static_assert` is to ensure a good compiler error message as close to the problem site as possible.

Reply:
New commits pushed and ready for re-review.

Reply:
Cool :)